addons/account/models/res_bank.py

```diff
@@ -47,15 +54,16 @@ def build_qr_code_vals(self, amount, free_communication, structured_communicatio
         return None
 
     def build_qr_code_url(self, amount, free_communication, structured_communication, currency, debtor_partner, qr_method=None, silent_errors=True):
-        candidate_method, amount, currency, debtor_partner, free_communication, structured_communication = \
-            self.build_qr_code_vals(amount, free_communication, structured_communication, currency, debtor_partner, qr_method, silent_errors)
-        return self._get_qr_code_url(candidate_method, amount, currency, debtor_partner, free_communication, structured_communication)
-
+        vals = self.build_qr_code_vals(amount, free_communication, structured_communication, currency, debtor_partner, qr_method, silent_errors)
+        if vals:
+            return self._get_qr_code_url(**vals)
```

Comment:
`_get_qr_code_url` has only positional arguments, not keywords args.. and the unpacking of a dictionary as positional argument should be done using a single *, but don't do that because odoo 14.0 requires only python 3.6 and iterating on a dictionary isn't officially guaranteed before 3.7

So you'll need to explicitly gives each parameter from `build_qr_code_vals` to the sub methods as kwargs

Reply:
Done :+1: Thanks!

Reply:
looks like it, if the runbot likes it too I'll r+